File: C/3053.c
#include <stdio.h>

int main() {
    int n;
    scanf("%d", &n);
    
    int posicao;  // 0 para 'A', 1 para 'B', 2 para 'C'
    char ini;
    scanf(" %c", &ini);

    if (ini == 'A') {
        posicao = 0;
    } else if (ini == 'B') {
        posicao = 1;
    } else {
        posicao = 2;
    }

    for (int i = 0; i < n; i++) {
        char jogada;
        scanf(" %c", &jogada);
        
        if (jogada == '1' && posicao != 2) {
            posicao = 1 - posicao;  // Troca entre 0 e 1
        } else if (jogada == '2' && posicao != 0) {
            posicao = 3 - posicao;  // Troca entre 1 e 2
        } else if (jogada == '3' && posicao != 1) {
            posicao = 2 - posicao;  // Troca entre 0 e 2
        }
    }

    if (posicao == 0) {
        printf("A\n");
    } else if (posicao == 1) {
        printf("B\n");
    } else {
        printf("C\n");
    }

    return 0;
}

File: Python/3053.py
n = int(input())
copos = [0,0,0]
ini = input()

if ini == 'A': copos[0] = 1
elif ini == 'B': copos[1] = 1
else: copos[2] = 1

for i in range(n):
    jogada = input()
    if jogada == '1':
        copos[0], copos[1] = copos[1], copos[0]
    elif jogada == '2':
        copos[1], copos[2] = copos[2], copos[1]
    else:
        copos[0], copos[2] = copos[2], copos[0]
        
if copos[0] == 1: print('A')
elif copos[1] == 1: print('B')
else: print('C')